nmdc_schema/migrators/migrator_from_X_to_PR19_and_PR70.py

```diff
@@ -0,0 +1,101 @@
+        >>> m = Migrator(adapter=adapter)
+        >>> m.fill_instrument_set()
+        >>> result = adapter.get_document_having_value_in_field('instrument_set', 'name', value='12T FT-ICR MS')
+        >>> result['name']
+        '12T FT-ICR MS'
+        >>> result['vendor']
+        'bruker'
+        >>> result['model']
+        'solarix_12T'
+        >>> result['type']
+        'nmdc:Instrument'
+
+        """
+
+        instruments = load_yaml_asset("migrator_from_X_to_PR19_and_PR70/instrument_set.yaml")
+        for instrument in instruments:
+            
+            # Mint instrument id and insert into dict
+            instrument_id = self.mint_ids()
+            instrument["id"] = instrument_id
+
+            self.adapter.insert_document("instrument_set", instrument)
+
+    def transform_instrument_slot(self, omics_doc: dict):
+        r"""
+        Changes the instrument_name slot to be instrument_used, and replaces the name string value with an identifier for 
+        the instrument
+
+        >>> from nmdc_schema.migrators.adapters.dictionary_adapter import DictionaryAdapter
+        >>>
+        >>> database = {'instrument_set': [{'id': 'nmdc:inst-456', 'name': '12T FT-ICR MS', 'model':'solarix_12T', 'vendor':'bruker', 'type':'nmdc:Instrument'}]} 
+        >>> adapter = DictionaryAdapter(database=database)
+        >>> m = Migrator(adapter=adapter)
+        >>> m.transform_instrument_slot({'id': 'nmdc:omcp-123', 'instrument_name': '12T_FTICR_B'})
+        {'id': 'nmdc:omcp-123', 'instrument_used': 'nmdc:inst-456'}
+        """
+
+        if "instrument_name" in omics_doc:
+            existing_instrument_name = omics_doc["instrument_name"]
+            
+            instruments = load_yaml_asset("migrator_from_X_to_PR19_and_PR70/instrument_set.yaml")
+            allowed_instrument_names = [instrument["name"] for instrument in instruments]
+            
+            matches = difflib.get_close_matches(existing_instrument_name, allowed_instrument_names, n=1, cutoff=0.25)
+
+            if matches:
+
+                    # Match instrument_name with with instrument set and get corresponding instrument id
+                    match = matches[0]
+                    self.logger.info(f"instrument_name in {omics_doc['id']} is {omics_doc['instrument_name']} and will be matched to instrument with name {match}")
+                    
+                    instrument = self.adapter.get_document_having_value_in_field(
+                        collection_name="instrument_set", field_name="name", value=match
+                    )
+
+                    instrument_id = instrument.get("id")
+                    omics_doc["instrument_used"] = instrument_id
+                    del omics_doc["instrument_name"]
+        
+            else:
+                self.logger.error(f"The instrument_name {omics_doc['id']} has a value of {omics_doc['instrument_name']}, but did not have any matches with the cutoff set at 0.25")
```

Comment:
Reminder for @eecavanna to update the migration framework to do something (TBD) when these errors are logged.